saleor/graphql/checkout/mutations/checkout_create.py

```diff
@@ -159,6 +159,18 @@ class CheckoutCreateInput(BaseInputObjectType):
         description=("The checkout validation rules that can be changed."),
     )
 
+    metadata = NonNullList(
+        MetadataInput,
+        description="Checkout public metadata." + ADDED_IN_321,
+        required=False,
+    )
+
+    private_metadata = NonNullList(
```

Comment:
Adding anything to private_metadata should require proper permission.  `MANAGE_CHECKOUTS` in that case.

Reply:
done + added test for that